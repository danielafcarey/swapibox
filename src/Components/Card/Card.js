import React from 'react';
import PropTypes from 'prop-types';

const Card = (props) => {
  const {
    id,
    title,
    item1,
    item2,
    item3,
    item4,
    favorite,
    section,
    toggleFavorite
  } = props;

  let imgSrc;
  if (favorite) {
    imgSrc = require('../../images/lightsaber-red.png');
  } else {
    imgSrc = require('../../images/lightsaber-white.png');
  }

  return (
    <div className={`Card ${section}`}>
      <h3 className="title">{ title }</h3>
      <button 
        onClick={ () => toggleFavorite(id) }
      >
        <img 
          src={ imgSrc } 
          alt="Add to favorites"
        />
      </button>
      <div className="info-list">
        <h4 className="info">{ item1 }</h4>
        <h4 className="info">{ item2 }</h4>
        <h4 className="item3">{ item3 }</h4>
        <h4 className="item4">{ item4 }</h4>
      </div>
    </div>
  );
};

Card.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  item1: PropTypes.string.isRequired,
  item2: PropTypes.string.isRequired,
  item3: PropTypes.string.isRequired,
  item4: PropTypes.string,
  favorite: PropTypes.bool.isRequired,
  section: PropTypes.string.isRequired,
  toggleFavorite: PropTypes.func.isRequired
};

export default Card;
