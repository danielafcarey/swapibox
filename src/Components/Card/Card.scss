.Card {
  display: grid;
  grid-template-rows: 1fr 2fr;
  grid-template-columns: 4fr 1fr;
  color: white;
  background-color: rgba(43, 43, 43, 0.65);
  margin: 15px;
  width: 300px;
  height: 350px;
  border-radius: 8px;

  .title {
    grid-area: 1/1/2/2;
    font-size: 1.8rem;
    padding: 0 0 0 20px;
    letter-spacing: 1px;
    margin-bottom: 5px;
  }

  button {
    grid-area: 1/2/2/3;
    border: none;
    background: none;
    
    img {
      height: 78px;
    }

  }

  .info-list {
    grid-area: 2/1/3/3;
  }

  h4 {
    font-size: 1rem;
    letter-spacing: 1px;
    padding: 0 7px;
    margin: 15px 5px 0 15px;
  }

}
