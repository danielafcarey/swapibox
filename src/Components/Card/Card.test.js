import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      id: 'id',
      title: 'is a card',
      item1: 'item1',
      item2: 'item2',
      item3: 'item3',
      item4: 'item4',
      favorite: false,
      toggleFavorite: jest.fn()
    }
  })

  it('should match the snapshot', () => {
    const wrapper = shallow(<Card { ...mockProps } />)

    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot if a card is a favorite', () => {
    const wrapper = shallow(<Card { ...mockProps } favorite={true} />)

    expect(wrapper).toMatchSnapshot();
  })

  it('calls the toggleFavorite function with the card id when button is clicked', () => {
    const spy = jest.fn(); 
    const wrapper = shallow(<Card { ...mockProps } toggleFavorite={spy} />);

    wrapper.find('button').simulate('click');

    expect(spy).toBeCalledWith('id');
  })
})
