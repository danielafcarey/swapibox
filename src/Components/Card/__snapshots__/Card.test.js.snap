// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={false}
    id="id"
    item1="item1"
    item2="item2"
    item3="item3"
    item4="item4"
    title="is a card"
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="title"
>
          is a card
</h3>,
        <button
          onClick={[Function]}
>
          <img
                    alt="Add to favorites"
                    src="lightsaber-white.png"
          />
</button>,
        <div
          className="info-list"
>
          <h4
                    className="info"
          >
                    item1
          </h4>
          <h4
                    className="info"
          >
                    item2
          </h4>
          <h4
                    className="item3"
          >
                    item3
          </h4>
          <h4
                    className="item4"
          >
                    item4
          </h4>
</div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "is a card",
          "className": "title",
        },
        "ref": null,
        "rendered": "is a card",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Add to favorites"
            src="lightsaber-white.png"
/>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Add to favorites",
            "src": "lightsaber-white.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="info"
>
              item1
</h4>,
            <h4
              className="info"
>
              item2
</h4>,
            <h4
              className="item3"
>
              item3
</h4>,
            <h4
              className="item4"
>
              item4
</h4>,
          ],
          "className": "info-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item1",
              "className": "info",
            },
            "ref": null,
            "rendered": "item1",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item2",
              "className": "info",
            },
            "ref": null,
            "rendered": "item2",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item3",
              "className": "item3",
            },
            "ref": null,
            "rendered": "item3",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item4",
              "className": "item4",
            },
            "ref": null,
            "rendered": "item4",
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="title"
>
            is a card
</h3>,
          <button
            onClick={[Function]}
>
            <img
                        alt="Add to favorites"
                        src="lightsaber-white.png"
            />
</button>,
          <div
            className="info-list"
>
            <h4
                        className="info"
            >
                        item1
            </h4>
            <h4
                        className="info"
            >
                        item2
            </h4>
            <h4
                        className="item3"
            >
                        item3
            </h4>
            <h4
                        className="item4"
            >
                        item4
            </h4>
</div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "is a card",
            "className": "title",
          },
          "ref": null,
          "rendered": "is a card",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Add to favorites"
              src="lightsaber-white.png"
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Add to favorites",
              "src": "lightsaber-white.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="info"
>
                item1
</h4>,
              <h4
                className="info"
>
                item2
</h4>,
              <h4
                className="item3"
>
                item3
</h4>,
              <h4
                className="item4"
>
                item4
</h4>,
            ],
            "className": "info-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item1",
                "className": "info",
              },
              "ref": null,
              "rendered": "item1",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item2",
                "className": "info",
              },
              "ref": null,
              "rendered": "item2",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item3",
                "className": "item3",
              },
              "ref": null,
              "rendered": "item3",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item4",
                "className": "item4",
              },
              "ref": null,
              "rendered": "item4",
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Card should match the snapshot if a card is a favorite 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    favorite={true}
    id="id"
    item1="item1"
    item2="item2"
    item3="item3"
    item4="item4"
    title="is a card"
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="title"
>
          is a card
</h3>,
        <button
          onClick={[Function]}
>
          <img
                    alt="Add to favorites"
                    src="lightsaber-red.png"
          />
</button>,
        <div
          className="info-list"
>
          <h4
                    className="info"
          >
                    item1
          </h4>
          <h4
                    className="info"
          >
                    item2
          </h4>
          <h4
                    className="item3"
          >
                    item3
          </h4>
          <h4
                    className="item4"
          >
                    item4
          </h4>
</div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "is a card",
          "className": "title",
        },
        "ref": null,
        "rendered": "is a card",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Add to favorites"
            src="lightsaber-red.png"
/>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Add to favorites",
            "src": "lightsaber-red.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="info"
>
              item1
</h4>,
            <h4
              className="info"
>
              item2
</h4>,
            <h4
              className="item3"
>
              item3
</h4>,
            <h4
              className="item4"
>
              item4
</h4>,
          ],
          "className": "info-list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item1",
              "className": "info",
            },
            "ref": null,
            "rendered": "item1",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item2",
              "className": "info",
            },
            "ref": null,
            "rendered": "item2",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item3",
              "className": "item3",
            },
            "ref": null,
            "rendered": "item3",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "item4",
              "className": "item4",
            },
            "ref": null,
            "rendered": "item4",
            "type": "h4",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="title"
>
            is a card
</h3>,
          <button
            onClick={[Function]}
>
            <img
                        alt="Add to favorites"
                        src="lightsaber-red.png"
            />
</button>,
          <div
            className="info-list"
>
            <h4
                        className="info"
            >
                        item1
            </h4>
            <h4
                        className="info"
            >
                        item2
            </h4>
            <h4
                        className="item3"
            >
                        item3
            </h4>
            <h4
                        className="item4"
            >
                        item4
            </h4>
</div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "is a card",
            "className": "title",
          },
          "ref": null,
          "rendered": "is a card",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Add to favorites"
              src="lightsaber-red.png"
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Add to favorites",
              "src": "lightsaber-red.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="info"
>
                item1
</h4>,
              <h4
                className="info"
>
                item2
</h4>,
              <h4
                className="item3"
>
                item3
</h4>,
              <h4
                className="item4"
>
                item4
</h4>,
            ],
            "className": "info-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item1",
                "className": "info",
              },
              "ref": null,
              "rendered": "item1",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item2",
                "className": "info",
              },
              "ref": null,
              "rendered": "item2",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item3",
                "className": "item3",
              },
              "ref": null,
              "rendered": "item3",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "item4",
                "className": "item4",
              },
              "ref": null,
              "rendered": "item4",
              "type": "h4",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
